src/nspawn/nspawn.c

```diff
@@ -3029,14 +3035,15 @@ static int outer_child(
         if (r < 0)
                 return r;
 
-        if (!cg_ns_supported()) {
+        if (!arg_use_cgns && !cg_ns_supported()) {
```

Comment:
The condition should be  `!arg_use_cgns || !cg_ns_supported()`, shouldn't it?
I should execute the code below if cgroup namesapce is not supported OR if should not be used.

E.g. 
If I set `arg_use_cgns = false` (not to use cgroup namespace) and `cg_ns_supported()` returns true (because the kernel supports it) the code below is not executed but it should, right? Please correct me if I am wrong